### PR TITLE
Update: swarm support for ansible-traefik-docker

### PR DESCRIPTION
**Pull request by till**
- introduces new vars to control if we run on swarm
    - `traefik_use_swarm`
    - (optional) `traefik_swarm_network_opts` 
    - `traefik_compose_version` (must be set to `3` for Swarm)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,4 +11,4 @@ test: build-testimage
 		-v /var/run/docker.sock:/var/run/docker.sock \
 		-w /tmp/${ROLENAME} \
 		${TESTIMAGENAME} \
-		sudo molecule test
+		sudo molecule test --scenario-name swarm
```
Comment (rwos) [inline Makefile]: just so we don't forget it - this needs to be changed to "run all scenarios" (not sure how yet :D)

File: defaults/main.yml [hunk 2]
```diff
@@ -34,3 +36,7 @@ traefik_consul_enable: false
 traefik_consul_endpoint: "127.0.0.1:8500"
 traefik_consul_prefix: "traefik"
 traefik_consul_watch: true
+
+# Compose, or Swarm?
+use_swarm: false
+swarm_network_opts: "--attachable --driver=overlay"
```
Comment (rwos) [inline defaults/main.yml]: shouldn't we prefix those with "traefik_" or something? Not sure what the ansible best practices are in that regard

File: templates/docker-compose.yml.j2 [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # {{ ansible_managed }}
-version: '2'
+version: '{{ traefik_compose_version }}'
```
Comment (rwos) [inline templates/docker-compose.yml.j2:2]: this works, but just always setting it to `'3'` is maybe easier?

File: Makefile (hunk 1 above)
Comment (rwos) [reply Makefile]: `molecule test --all` does that apparently

File: templates/docker-compose.yml.j2 (hunk 3 above)
Comment (till) [reply templates/docker-compose.yml.j2:2]: @rwos I am not sure, theoretically yes, but it's also just a weird hack.

Comment (till) [reply templates/docker-compose.yml.j2:2]: Best would be to override the template completely, but I am not sure I want to add that right now.

File: molecule/swarm/tests/test_default.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
+    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('all')
+
+
+def test_traefik_is_running_on_swarm(host):
+    out = host.check_output(
+        'docker ps --filter "label=traefik.backend=traefik"'
+        + ' --format {%raw%}"{{.Image}}"{%endraw%}')
+    assert 'traefik:latest' == out
```
Comment (till) [inline molecule/swarm/tests/test_default.py:14]: This actually responds different on Swarm. Go figure. 🤦‍♂️

**Review by rwos — APPROVED** [review]

**Pull request merged**